atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/PersistentLockManager.java

```diff
@@ -62,25 +66,32 @@ public synchronized void shutdown() {
         log.info("Shutdown completed!");
     }
 
-    public synchronized void acquirePersistentLockWithRetry() {
+    // We don't synchronize this method, to avoid deadlocking if {@link #shutdown} is called while attempting
+    // to acquire a lock.
+    public void acquirePersistentLockWithRetry() {
+        while (!tryAcquirePersistentLock()) {
+            waitForRetry();
+        }
+    }
+
+    private synchronized boolean tryAcquirePersistentLock() {
         if (isShutDown) {
             // To avoid a race condition on shutdown, we don't want to acquire any more.
             log.info("The PersistentLockManager is shut down, and therefore rejected a request to acquire the lock.");
-            return;
+            return true;
```

Comment:
yeah this is weird.. the previous behavior was to silently return if we were shutdown, so this is the equivalent with the refactor.

it's probably fine to throw an exception here, as the only time this will ever happen is when the server is shutting down